webflib.py

```diff
@@ -19,29 +19,45 @@
 from configobj import ConfigObj
 
 
+class WebFactionDBUser(object):
+    def __init__(self, username, password, db_type):
+        super(WebFactionDBUser, self).__init__()
+        self.username = username
+        self.password = password
+        self.db_type = db_type
+                
 API_URL = 'https://api.webfaction.com/'
 CONF = os.path.expanduser('~/.webfrc')
-
 class WebFactionXmlRpc(object):
 
     '''WebFaction XML-RPC server proxy class'''
 
-    def __init__(self, user, password):
-        self.log = logging.getLogger('webf')
+    def __init__(self, user=None, password=None, machine=None):
+        self.log = logging.getLogger("webf")
         self.session_id = None
         self.server = None
+        if not (user and password):
+            try:
+                user, password = WebFactionXmlRpc.get_config()
+            except NotImplementedError as e:
```

Comment:
You can do this a little more cleanly by using `exception`, which will automatically handle the traceback info for you, e.g.,

```
except NotImplementedError:
    self.log.exception("Could not find configuration")
```